controllers/components/kueue/kueue_controller.go

```diff
@@ -56,6 +57,18 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 		Owns(&promv1.PrometheusRule{}).
 		Owns(&admissionregistrationv1.MutatingWebhookConfiguration{}).
 		Owns(&admissionregistrationv1.ValidatingWebhookConfiguration{}).
+		// We need to dynamically "own" these 2 resource types due to OCP version differences
+		// TODO: once we do not support 4.17-, could move it to :
+		// Owns(&admissionregistrationv1.ValidatingAdmissionPolicy{}).
+		// Owns(&admissionregistrationv1.ValidatingAdmissionPolicyBinding{}).
+		WatchesGVK(
+			gvk.ValidatingAdmissionPolicy,
+			reconciler.Dynamic(),
```

Comment:
You probably need to add a predicate to the `reconciler.Dynamic()` function as those are supported only on 4.16+

```go
reconciler.Dynamic(func(ctx context.Context, request *types.ReconciliationRequest) bool {
    return Is_OCP_4_16+()
})
```

Reply:
yep, only tested on 4.17, still need to fix it for 4.17-